fedimint-server/src/net/api/http_auth.rs

```diff
@@ -0,0 +1,159 @@
+use std::error::Error as StdError;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use anyhow::bail;
+use base64::engine::general_purpose::STANDARD;
+use base64::Engine;
+use fedimint_logging::LOG_NET_AUTH;
+use futures::{Future, FutureExt as _, TryFutureExt as _};
+use http::HeaderValue;
+use hyper::body::Body;
+use hyper::{http, Request, Response};
+use tower::Service;
+use tracing::{debug, info};
+
+#[derive(Clone, Debug)]
+pub struct HttpAuthLayer {
+    // surprisingly, a new `HttpAuthService` is created on every http request, so to avoid
+    // cloning every element of the vector, we pre-compute and `Arc` the whole thing
+    auth_base64: Arc<Vec<String>>,
+}
+
+impl HttpAuthLayer {
+    pub fn new(secrets: Vec<String>) -> Self {
+        if secrets.is_empty() {
+            info!(target: LOG_NET_AUTH, "Api available for public access");
+        } else {
+            info!(target: LOG_NET_AUTH, num_secrets = secrets.len(), "Api available for private access");
+        }
+        Self {
+            auth_base64: secrets
+                .iter()
+                .map(|p| STANDARD.encode(format!("fedimint:{p}")))
```

Comment:
The `fedimint:<secret>` format will allow us to eventually expand the supported auth schemes to blinded tokens etc.